test/e2e/resources_test.go

```diff
@@ -87,31 +87,39 @@ func TestCustomResourcesLimits(t *testing.T) {
 		return client.Do(req)
 	}
 
-	pokeCowForMB := func(mb int) error {
+	bloatAndCheck := func(mb int, wantSuccess bool) {
+		expect := "failure"
+		if wantSuccess {
+			expect = "success"
+		}
+		klog.V(2).Infof("Bloating by %d MB and expecting %s.\n", mb, expect)
 		u, _ := url.Parse(endpoint.String())
 		q := u.Query()
 		q.Set("bloat", fmt.Sprintf("%d", mb))
 		u.RawQuery = q.Encode()
 		response, err := sendPostRequest(test.ServingFlags.ResolvableDomain, u)
 		if err != nil {
-			return err
-		}
-		if response.StatusCode != http.StatusOK {
-			return fmt.Errorf("StatusCode = %d, want %d", response.StatusCode, http.StatusOK)
+			klog.V(5).Infof("Received error '%+v' from sendPostRequest (may be expected)\n", err)
+			if wantSuccess {
+				t.Fatalf("Didn't get a response from bloating RAM with %d MBs", mb)
+			}
+		} else if response.StatusCode == http.StatusOK {
+			if !wantSuccess {
+				t.Fatalf("We shouldn't have got a response from bloating RAM with %d MBs", mb)
+			}
+		} else if response.StatusCode == http.StatusBadRequest {
+			t.Error("Test Issue: Received BadRequest from test app, which probably means the test & test image are not cooperating with each other.")
```

Comment:
This error seems unnecessarily long and should include the request that was sent to make debugging easier.

Reply:
Do you think this information should not be provided in the log -- like move the second half to a comment?

Reply:
As for including request, if people run at verbosity 6 through 9 they can see the request, so is that not sufficient?